includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -1166,13 +1166,16 @@ public function process_payment_with_saved_payment_method( $order_id, $can_retry
 				];
 				if ( false === $intent ) {
 					$request['confirm'] = 'true';
+
 					// SEPA setup intents require mandate data.
+					$ip_address = WC_Geolocation::get_ip_address();
```

Comment:
Optional: since much of the code is the same as the above, we could consider moving this to a new shared method.

Reply:
@wjrosa Good suggestion! Added in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3900/commits/d2cc6b148288c690489a5ada2ff857631d920856.

Could you give it another 👀, to make sure the refactor looks good?

Reply:
Sorry, I missed this ping. The changes look great! Thanks for working on it!